spring-boot-project/spring-boot-docs/src/docs/asciidoc/messaging/pulsar.adoc

```diff
@@ -0,0 +1,132 @@
+[[messaging.pulsar]]
+== Apache Pulsar Support
+https://pulsar.apache.org/[Apache Pulsar] is supported by providing auto-configuration of the {spring-pulsar-docs}[Spring for Apache Pulsar] project.
+
+Spring Boot will auto-configure and register the classic (imperative) Spring Pulsar components when `org.springframework.pulsar:spring-pulsar` is on the classpath.
+It will do the same for the reactive components when `org.springframework.pulsar:spring-pulsar-reactive` is on the classpath.
+
+There are `spring-boot-starter-pulsar` and `spring-boot-starter-pulsar-reactive` "`Starters`" for conveniently collecting the dependencies for imperative and reactive use, respectively.
+
+TIP: The Reactive auto-configuration relies on the imperative auto-configuration and therefore the reactive starter includes the imperative starter.
+As such, the imperative components are auto-configured and available in both reactive and imperative scenarios.
+However, using the imperative components in a fully reactive app is not recommended as it could cause blocking issues.
```

Comment:
Not as it is currently written. It would be quite the effort to do this. The reactive components leverage the reactive client which adapts the original client (which is imperative and auto-configured). I believe there are other technologies auto-configured in Boot whose reactive components leverage the imperative ones, no?

Reply:
> The reactive components leverage the reactive client which adapts the original client (which is imperative and auto-configured)

I wonder if the original, imperative client has to be a bean in the reactive case. Perhaps it does for ease of configuration property binding and the like. Not something to worry about for this PR, but probably worth revisiting down the road before 3.2 GAs.

>  there are other technologies auto-configured in Boot whose reactive components leverage the imperative ones, no?

We have some situations where there's a low-level bean that's shared between imperative and reactive (Elasticsearch client transport and the ElasticsearchClient and ReactiveElasticsearchClient that use it, for example) but I can't think of a situation where we have an imperative bean that we expect applications to use extensively that's then wrapped by a reactive bean. I may well be forgetting something though.

Reply:
> I wonder if the original, imperative client has to be a bean in the reactive case. Perhaps it does for ease of configuration property binding and the like. Not something to worry about for this PR, but probably worth revisiting down the road before 3.2 GAs.

I'm happy to take a look and revisit in that timeframe.